graph.h

```diff
@@ -1,4 +1,9 @@
+#ifndef GRAPH_H
+#define GRAPH_H
+
 #include <complex.h>
+#include <cairo/cairo.h>
```

Comment:
including the headers here allows the types to be used and avoids requiring the user to have to know to include them in his code.